optuna_dashboard/ts/components/GraphTimeline.tsx

```diff
@@ -93,11 +115,20 @@ const plotTimeline = (trials: Trial[], mode: string) => {
   }
 
   const makeTrace = (bars: Trial[], state: string, color: string) => {
+    const isRunning = state === runningKey
     const starts = bars.map((b) => b.datetime_start ?? new Date())
-    const completes = bars.map((b, i) => b.datetime_complete ?? starts[i])
+    const runDurations = bars.map((b) => {
+      const start = b.datetime_start?.getTime() ?? new Date().getTime()
+      const complete = b.datetime_complete?.getTime() ?? start
+      // By using 1 as the min value, we can recognize these bars at least when zooming in.
+      return Math.max(
+        1,
+        !isRunning ? complete - start : maxDatetime.getTime() - start
+      )
+    })
```

Comment:
[question] Why do we need to get max here? If we fix the code as I said, those lines can be just `complete - start`?

Reply:
For me, it seems that we can use the following lines:

```
    const runDurations = bars.map((b, i) => {
      const start = starts[i].getTime()
      const complete = b.datetime_complete?.getTime() ?? maxDatetime.getTime()
      return complete - start
    }
```

Reply:
The lower bound `1` is necessary to be able to recognize at least bars exist in case `runDurations=0`.
Otherwise, I think your code is better.

Reply:
Oh, actually, `const complete = b.datetime_complete?.getTime() ?? maxDatetime.getTime()` is incorrect when we think of `trialState===WAITING`.

Reply:
Ok, I understand that you want to show `waiting` and  in some cases also `running` trials.

In the optuna, it uses the current time for complete if a trial does not have it when it is `waiting` or `running`.
ref: https://github.com/optuna/optuna/blob/master/optuna/visualization/_timeline.py#L85-L86 Are you planning to make back port to optuna itself?

Reply:
I did the backport [here](https://github.com/optuna/optuna/pull/5168)